lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -168,6 +217,171 @@ def partition_pdf(
+
+        deformable_layout = []
+        if tracemalloc.is_tracing():
+            before = tracemalloc.take_snapshot()
+        for i in convert_from_path_streamed_batched(filename, batch_size):
+            parts = self.process_batch(
+                i,
+                threshold=threshold,
+                use_ocr=use_ocr,
+                ocr_images=ocr_images,
+                ocr_tables=ocr_tables,
+                extract_table_structure=extract_table_structure,
+                table_structure_extractor=table_structure_extractor,
+                extract_images=extract_images,
+            )
+            assert len(parts) == len(i)
+            deformable_layout.extend(parts)
+            if tracemalloc.is_tracing():
+                gc.collect()
+                after = tracemalloc.take_snapshot()
+                top_stats = after.compare_to(before, "lineno")
+
+                print("[ Top 10 differences ]")
+                for stat in top_stats[:10]:
+                    print(stat)
+                before = after
+                display_top(after)
+
+        if pdfminer is not None:
+            with LogTime("wait_for_pdfminer", log_start=True):
+                pdfminer_layout = pdfminer.result()
+            assert len(pdfminer_layout) == len(deformable_layout), f"{len(pdfminer_layout)} vs {len(deformable_layout)}"
+            with LogTime("pdfminer_supplement"):
+                for d, p in zip(deformable_layout, pdfminer_layout):
+                    self._supplement_text(d, p)
+
+        if tracemalloc.is_tracing():
+            (current, peak) = tracemalloc.get_traced_memory()
+            logging.info(f"Memory Usage current={current} peak={peak}")
+            top = tracemalloc.take_snapshot()
+            display_top(top)
+        return deformable_layout
+
+    @staticmethod
+    def _run_pdfminer(pdf_path):
+        pdfminer = PDFMinerExtractor()
+        with LogTime("pdfminer_extract", log_start=True):
+            pdfminer_layout = pdfminer.extract(pdf_path)
+
+        return pdfminer_layout
+
+    def gc_tensor_dump(self):
+        if not tracemalloc.is_tracing():
+            return
+        import torch
+        import gc
+
+        count = 0
+        for obj in gc.get_objects():
+            try:
+                if torch.is_tensor(obj) or (hasattr(obj, "data") and torch.is_tensor(obj.data)):
+                    # print(type(obj), obj.size())
+                    count = count + 1
+            except Exception as e:
+                print(f"Exception {e}")
+        print(f"Found {count} tensors")
+
+    def process_batch(
+        self,
+        batch: list[Image.Image],
+        threshold,
+        use_ocr,
+        ocr_images,
+        ocr_tables,
+        extract_table_structure,
+        table_structure_extractor,
+        extract_images,
+    ) -> Any:
+        with LogTime("infer"):
+            deformable_layout = self.model.infer(batch, threshold, None)
```

Comment:
How does this differ from the sequential processing? that batchify does not work?

Reply:
I don't understand the question. The net effect of sequential and batched processing should be the same, but how they get there is different.

The sequential processing does:
1. pdf->list[images]
2. list[images]->RGB list[images]
3. for all images, infer(image batch)
4. run ocr or pdfminer and join
5. for each page, extract table structure and images

The batched processing does:
1. Start pdfminer if needed
2. Start incremental pdf -> images conversion
3. For each image batch:
4. .... run infer
5. .... extract ocr if requested
6. .... extract table structure if requested
7. .... extract images if requested
8. If using pdfminer, wait for completion and join

_process_batch is that inner loop (steps 4-7)

Reply:
The sequence process also do batch infer. So the batching here refers to, table/image extraction and orc are batched?